### PR TITLE
Add multiple binding free shipping amounts

### PR DESCRIPTION
**Pull request by razvanudream**
#### What does this PR do? \*

This PR handles the case when the store has multiple bindings and wants to set a free shipping amount for each individual binding. With the current version, the free shipping amount in the app's settings would apply to all bindings.

Also, it hides the text1 and text2 above the completion bar because it's not OK to display "Complete your order to have FREE SHIPPING" when the free shipping amount has already been hit.  If you'd rather have me add a boolean prop as well to allow some stores to keep these messages up there regardless of whether or not the shipping amount has been hit, let me know

#### How to test it? \*

The app settings has two extra options

isMultiBindingStore boolean which if checked tells the code to default to the values inside the freeShippingAmounts array
freeShippingAmounts array which consists of objects with bindingId and freeShippingAmount for each object

I've attached some screenshots below as well to see the difference

This one is for B2B binding - I've set the amount to 352.1

image

This one is for B2C binding - I've set the amount to 252.1

image

#### Describe alternatives you've considered, if any. \*

No alternatives as this is a feature the app didn't have.  Conditional layout again wasn't an option because the free shipping amount is taken from the app's settings so it had to be done at app level, not at store-theme level.

### CODE REVIEW

File: react/components/MinicartFreeshipping.tsx
Comment (edyespinal) [inline react/components/MinicartFreeshipping.tsx:43]: I might be wrong, but why are we defaulting to `0` here?
Won't this mean that, if a user doesn't set the free shipping amount for every binding of the store, the bindings that are not configured will get free shipping starting from `0`.

If a store is not configured on the app settings, they should either not have free shipping or default to the value set when multi binding is unchecked.

**Review by edyespinal — CHANGES_REQUESTED** [review]

File: manifest.json
Comment (razvanudream) [reply manifest.json:48]: Ah nice, I didn't know about that option, makes the code a bit cleaner. I pushed that change.

File: react/components/MinicartFreeshipping.tsx
Comment (razvanudream) [reply react/components/MinicartFreeshipping.tsx]: You're correct. I adapted the code and tested the following scenarios

1. If freeShippingAmount is set AND bindingBounded is not => use freeShippingAmount -> OK
2. bindingBounded is set AND the freeShippingAmount is set for the current binding => use the freeShippingAmount for that binding -> OK (I had to do the binding check twice 2 determine whether or not it is set, otherwise it would get to the main logic and would cause the amount to be set as 0 AKA initial state)

The reason for the if on line 46 is because find can return undefined as well and react doesn't like that. 

I've tested several other scenarios and everything seems to be covered.

File: react/components/MinicartFreeshipping.tsx
Comment (edyespinal) [inline react/components/MinicartFreeshipping.tsx:19]: I think that since we are extending `Settings` there's no need to have `freeShippingAmount` here.

**Review by edyespinal — COMMENTED** [review]
Great work!
Everything looks good and there is just one minor change I would make before merging.

File: react/components/MinicartFreeshipping.tsx
Comment (razvanudream) [reply react/components/MinicartFreeshipping.tsx]: Ah yes, you're right. Removed that one from the interface. Cheers

**Review by edyespinal — APPROVED** [review]

**Comment by razvanudream** [comment]
Thanks @edyespinal - is there anything that needs to be done in order for this to be merged & released?   I'm waiting for this release to be able to finish a task for one of our clients this sprint.  Cheers!

**Pull request merged**
